packages/sanity/src/core/form/studio/assetSourceMediaLibrary/shared/MediaLibrarySource.tsx

```diff
@@ -43,23 +43,16 @@ const MediaLibraryAssetSource = function MediaLibraryAssetSource(
     if (libraryIdProp || fetchedLibraryId) {
       return
     }
-    client.request({uri: `/projects/${projectId}`}).then((project) => {
-      const {organizationId} = project
-      client
-        .withConfig({
-          useProjectHostname: false,
-        })
-        .request({uri: `/media-libraries`, query: {organizationId}, useGlobalApi: true})
-        .then((result) => {
-          const libraryIdFromResult = result.data[0]?.id
-          if (libraryIdFromResult) {
-            // Add to cache for this project (organization)
-            if (projectId) {
-              fetchedLibraryIdCache.set(projectId, libraryIdFromResult)
-            }
-            setFetchedLibraryId(libraryIdFromResult)
-          }
-        })
+    if (!projectId) {
+      throw new Error('projectId is required to fetch Media Library ID')
+    }
```

Comment:
`projectId` is derived from the client, so the chance of hitting this error is about zero.

I do think throwing is correct here though as opposed to early return or if statement.